### PR TITLE
Redux store to display books

### PR DESCRIPTION
**Pull request by OpondoG**
## In this pull request I:
- [x] First of all - make sure that you keep state immutable in reducers!
- [x] Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
- [x] Import Redux Provider and your store in the main component of your React app.
- [x] Connect your app to the Redux store.
- [x] Make sure that list of books displays books from Redux store.
- [x] Use useSelector to consume the state.
- [x] In your React component responsible for adding new books, add functionality for "Add new" button click:
- [x] Import Redux useDispatch.
- [x] Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
- [x] Takes the inputs from the state and generate a unique id and add them to an object.
- [x] Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action - [x] - [x] Creator that you created before and use it here).
- [x] In your React component responsible for removing books, add a functionality for "Remove" button click:
Import Redux useDispatch.
- [x] Dispatch a corresponding action (import Action Creator that you created before and use it here).
- [x] In your React component responsible for categories, add a functionality for "Check status" button click:
- [x] Import Redux useDispatch.
- [x] Dispatch a corresponding action (import Action Creator that you created before and use it here).
- [x] Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Make sure that you used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### CODE REVIEW

File: src/Pages/category.js [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+    <div>
+      <h1>Category</h1>
+      <button
+        type="button"
+        onClick={() => {
+          dispatch(checkStatus());
+        }}
+      >
+        check status
+      </button>
```
Comment (m05e5) [inline src/Pages/category.js]: - [x] As mentioned in the requirements, the check status button is supposed to dispatch the message and display it but currently, the button is dispatching the message from redux but it's not displaying the 'Under construction', please kindly fix that

**Review by m05e5 — COMMENTED** [review]
Hi @OpondoG , 

Good job so far. I really like what you did!
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve
- [x] Please check the comments bellow


_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Pages/category.js [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
-    <button type="button"> check status</button>
-  </>
-);
+const Category = () => {
+  const res = useSelector((state) => state.categories);
+  const dispatch = useDispatch();
+  return (
+    <div>
+      <span>{res}</span>
+      <button type="button" onClick={() => dispatch(checkStatus())}>Check Status</button>
```
Comment (muneebulrehman) [inline src/Pages/category.js:11]: Optional.
It would be better if you would hide the button once the message is displayed.

**Review by muneebulrehman — APPROVED** [review]
Hi @OpondoG , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Good follow-up with the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**